reference-implementation/lib/transform-stream.js

```diff
@@ -379,23 +435,26 @@ class TransformStream {
 
     this._transformStreamController = new TransformStreamDefaultController(this);
 
+    this._readableType = transformer.readableType;
```

Comment:
Is there a reason not to validate the value of `readableType` and `writableType` here? It seems weird to me.